src/shared/server-api/api-request.js

```diff
@@ -0,0 +1,62 @@
+function getCookie(name) {
```

Comment:
@wlach I'm assuming we want devs to only use this to craft new API functions, not use these directly, correct?

@bcolloran you had mentioned before that we might want to put a note or something to make it very clear that using functions like these directly is a no-go. Any thoughts on how we might handle that here?

Reply:
hmm well in this case the function is not exported, so it's "private" within the module. iirc the case that came up the other day was about a function that you were exporting, but which you said should not be used in other code?

since this is not exported, it's a bit different, but the risk remains that someone will forget about it and export it down the road. i think it would be good to mark things like this that we want to be treated as "private" _somehow_.

unlike in python, the js community seems to frown on the underscore prefix to indicate that a variable should be treated as private, but i'm not convinced by the arguments i've seen. i would be ok with adopting that convention (and i'm also open to other ideas). what do you guys think?

Reply:
I was referring to the other functions in this file which are exported, not this one. Seems to me the model we want is for devs to use this to create new APIs (eg the file or notebook apis will made in this PR) not directly

Reply:
I don't have a strong feeling one way or another on this. We coudl at the very least put a big comment at the top saying this file should not be used directly.

Reply:
Yeah, I don't think javascript really has any methods to enforce private methods. 

I think a simple comment like this at the top of the file should do the trick:

```js
// Please do not use any of these methods directly in your code! Instead, create
// methods over them, as is done elsewhere in this directory. This will make it easier to 
// switch to different authentication methods in the future.
```